Lib/test/test_veronapy.py

```diff
@@ -547,5 +547,18 @@ def test_global_dict_mutation(self):
         self.assertTrue(isimmutable(f1))
         self.assertRaises(NotWriteableError, f1)
 
+class TestPoolAllocation(unittest.TestCase):
+    # If pooling does not reset region between allocations,
+    # then the second call to f will result in `a` being owned by
+    # the first region that no has been deallocated.  This
+    # will result in a UAF that ASAN can detect.
+    def test_pool_allocation(self):
+        def f():
+            r = Region()
+            a = {}
+            r.add_object(a)
```

Comment:
You could actually write this as `r.f = {}` because the code that creates the internal dictionary in `r` uses the write barrier to take ownership over `{}` in this case. This would probably make this test more future proof.

Reply:
Are we expecting `add_object` to go away?  I like `add_object` from a test point of view as it isn't also testing the write barrier logic.

Reply:
We are not expecting it to go away until we are done, I think!

Reply:
(At least I haven't considered removing it.)